src/probnum/linalg/solvers/_state.py

```diff
@@ -31,20 +31,23 @@ def __init__(
         prior: "probnum.linalg.solvers.beliefs.LinearSystemBelief",
         rng: Optional[np.random.Generator] = None,
     ):
+        self.rng: Optional[np.random.Generator] = rng
```

Comment:
Do we want this to be part of the state? Considering the `ad-prototype` I would suggest passing it to the `solve_iterator` and forward it to the component which requires an rng, e.g. the policy.